### PR TITLE
Get next node custom

### PR DESCRIPTION
**Pull request by GabrielLyonB**
## Objective
- Include methods to retrieve a node or a nodable by a custom nodable attribute instead of only the first one.

## Description
- I implemented methods in the `QueueIt::Queue` class to obtain the node searching by the nodable attribute for diferent queue lengths.
- I implemented the corresponding methods in the `QueueIt::Queable` concern to obtain the node or the nodable by a custom nodable attribute.
- I included tests for both classes.

### CODE REVIEW

File: app/models/queue_it/queue.rb [hunk 1]
```diff
@@ -27,21 +27,41 @@ def two_nodes?
...
+        nodes.where.not(kind: :any).find_each { |node| node.update!(kind: :any) }
         old_head_node.update!(kind: :tail, parent_node: old_tail_node)
         old_tail_node.update!(kind: :head, parent_node: nil)
         old_head_node
       end
-      next_node
+    end
+
+    def get_next_by_in_generic_queue(nodable_attribute, attribute_value)
+      return get_next_in_queue_generic if head_node.nodable.send(nodable_attribute) == attribute_value
```
Comment (monkeyci) [inline app/models/queue_it/queue.rb]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Layout/LineLength: Line is too long. [102/100]

File: app/models/queue_it/queue.rb [hunk 2]
```diff
@@ -27,21 +27,41 @@ def two_nodes?
...
+
+      current_node = head_node.child_node
+      while !(current_node.nodable.send(nodable_attribute) == attribute_value && current_node != tail_node)
```
Comment (monkeyci) [inline app/models/queue_it/queue.rb]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Layout/LineLength: Line is too long. [107/100]

File: spec/concerns/queue_it/queable_spec.rb [hunk 3]
```diff
@@ -148,6 +148,146 @@
...
+      let(:in_head) { true }
+
+      before do
+        task.push_to_queue(first_nodable)
+        task.push_to_queue(second_nodable, in_head)
+      end
+
+      context 'when searching for tail node' do
+        it "expects to return first_nodable and to not change nodables kind's" do
+          expect(task.get_next_node_in_queue_by('id', first_nodable.id).nodable).to eq(first_nodable)
```
Comment (monkeyci) [inline spec/concerns/queue_it/queable_spec.rb]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Layout/LineLength: Line is too long. [101/100]

File: spec/concerns/queue_it/queable_spec.rb [hunk 4]
```diff
@@ -148,6 +148,146 @@
...
+          expect(task.queue.nodes.find_by(kind: :tail).nodable).to eq(first_nodable)
+          expect(task.queue.nodes.find_by(kind: :head).nodable).to eq(second_nodable)
+        end
+      end
+
+      context 'when searching for head node' do
+        it "expects to return second_nodable node and to change node 'head' for 'tail' and reverse" do
```
Comment (monkeyci) [inline spec/concerns/queue_it/queable_spec.rb]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Layout/LineLength: Line is too long. [102/100]

File: spec/concerns/queue_it/queable_spec.rb [hunk 5]
```diff
@@ -148,6 +148,146 @@
...
+          expect(task.get_next_node_in_queue_by('id', second_nodable.id).nodable).to eq(second_nodable)
```
Comment (monkeyci) [inline spec/concerns/queue_it/queable_spec.rb]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Layout/LineLength: Line is too long. [103/100]

File: app/models/concerns/queue_it/queable.rb [hunk 6]
```diff
@@ -37,20 +54,13 @@ def get_next_node_in_queue
...
       return if local_queue.empty?
+      return [local_queue.head_node.nodable.send(nodable_attribute)] if local_queue.one_node?
 
-      if local_queue.one_node?
-        [local_queue.head_node.nodable.send(nodable_attribute)]
-      elsif local_queue.two_nodes?
+      if local_queue.two_nodes?
         [local_queue.head_node.nodable.send(nodable_attribute),
-          local_queue.tail_node.nodable.send(nodable_attribute)]
+         local_queue.tail_node.nodable.send(nodable_attribute)]
```
Comment (aaguilarg) [inline app/models/concerns/queue_it/queable.rb]: Why not use `formatted_generic_queue` here?

**Review by aaguilarg — COMMENTED** [review]
Looks good to me! 🔥 💯 
Just one little question.

File: app/models/concerns/queue_it/queable.rb (hunk 6 above)
Comment (GabrielLyonB) [reply app/models/concerns/queue_it/queable.rb]: Code legacy. I'll updated.

**Review by aaguilarg — APPROVED** [review]

**Pull request merged**